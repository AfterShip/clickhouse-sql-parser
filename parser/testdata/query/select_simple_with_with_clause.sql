WITH
    cte1 AS (SELECT f1 FROM t1),
    cte2 AS (SELECT f2 FROM t2)
SELECT
    cte1.f1,
    cte2.f2,
    t3.f3
FROM
    t3,cte1,cte2

