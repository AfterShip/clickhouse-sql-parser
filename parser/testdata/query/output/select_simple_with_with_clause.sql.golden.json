[
  {
    "SelectPos": 0,
    "StatementEnd": 124,
    "With": {
      "WithPos": 0,
      "EndPos": 46,
      "CTEs": [
        {
          "CTEPos": 9,
          "Expr": {
            "Name": "cte1",
            "QuoteType": 1,
            "NamePos": 9,
            "NameEnd": 13
          },
          "Alias": {
            "SelectPos": 18,
            "StatementEnd": 35,
            "With": null,
            "Top": null,
            "HasDistinct": false,
            "SelectItems": [
              {
                "Expr": {
                  "Name": "f1",
                  "QuoteType": 1,
                  "NamePos": 25,
                  "NameEnd": 27
                },
                "Modifiers": [],
                "Alias": null
              }
            ],
            "From": {
              "FromPos": 28,
              "Expr": {
                "Table": {
                  "TablePos": 33,
                  "TableEnd": 35,
                  "Alias": null,
                  "Expr": {
                    "Database": null,
                    "Table": {
                      "Name": "t1",
                      "QuoteType": 1,
                      "NamePos": 33,
                      "NameEnd": 35
                    }
                  },
                  "HasFinal": false
                },
                "StatementEnd": 35,
                "SampleRatio": null,
                "HasFinal": false
              }
            },
            "ArrayJoin": null,
            "Window": null,
            "Prewhere": null,
            "Where": null,
            "GroupBy": null,
            "WithTotal": false,
            "Having": null,
            "OrderBy": null,
            "LimitBy": null,
            "Limit": null,
            "Settings": null,
            "Format": null,
            "UnionAll": null,
            "UnionDistinct": null,
            "Except": null
          }
        },
        {
          "CTEPos": 42,
          "Expr": {
            "Name": "cte2",
            "QuoteType": 1,
            "NamePos": 42,
            "NameEnd": 46
          },
          "Alias": {
            "SelectPos": 51,
            "StatementEnd": 68,
            "With": null,
            "Top": null,
            "HasDistinct": false,
            "SelectItems": [
              {
                "Expr": {
                  "Name": "f2",
                  "QuoteType": 1,
                  "NamePos": 58,
                  "NameEnd": 60
                },
                "Modifiers": [],
                "Alias": null
              }
            ],
            "From": {
              "FromPos": 61,
              "Expr": {
                "Table": {
                  "TablePos": 66,
                  "TableEnd": 68,
                  "Alias": null,
                  "Expr": {
                    "Database": null,
                    "Table": {
                      "Name": "t2",
                      "QuoteType": 1,
                      "NamePos": 66,
                      "NameEnd": 68
                    }
                  },
                  "HasFinal": false
                },
                "StatementEnd": 68,
                "SampleRatio": null,
                "HasFinal": false
              }
            },
            "ArrayJoin": null,
            "Window": null,
            "Prewhere": null,
            "Where": null,
            "GroupBy": null,
            "WithTotal": false,
            "Having": null,
            "OrderBy": null,
            "LimitBy": null,
            "Limit": null,
            "Settings": null,
            "Format": null,
            "UnionAll": null,
            "UnionDistinct": null,
            "Except": null
          }
        }
      ]
    },
    "Top": null,
    "HasDistinct": false,
    "SelectItems": [
      {
        "Expr": {
          "Database": null,
          "Table": {
            "Name": "cte1",
            "QuoteType": 1,
            "NamePos": 81,
            "NameEnd": 85
          },
          "Column": {
            "Name": "f1",
            "QuoteType": 1,
            "NamePos": 86,
            "NameEnd": 88
          }
        },
        "Modifiers": [],
        "Alias": null
      },
      {
        "Expr": {
          "Database": null,
          "Table": {
            "Name": "cte2",
            "QuoteType": 1,
            "NamePos": 94,
            "NameEnd": 98
          },
          "Column": {
            "Name": "f2",
            "QuoteType": 1,
            "NamePos": 99,
            "NameEnd": 101
          }
        },
        "Modifiers": [],
        "Alias": null
      },
      {
        "Expr": {
          "Database": null,
          "Table": {
            "Name": "t3",
            "QuoteType": 1,
            "NamePos": 107,
            "NameEnd": 109
          },
          "Column": {
            "Name": "f3",
            "QuoteType": 1,
            "NamePos": 110,
            "NameEnd": 112
          }
        },
        "Modifiers": [],
        "Alias": null
      }
    ],
    "From": {
      "FromPos": 113,
      "Expr": {
        "JoinPos": 122,
        "Left": {
          "Table": {
            "TablePos": 122,
            "TableEnd": 124,
            "Alias": null,
            "Expr": {
              "Database": null,
              "Table": {
                "Name": "t3",
                "QuoteType": 1,
                "NamePos": 122,
                "NameEnd": 124
              }
            },
            "HasFinal": false
          },
          "StatementEnd": 124,
          "SampleRatio": null,
          "HasFinal": false
        },
        "Right": {
          "JoinPos": 125,
          "Left": {
            "Table": {
              "TablePos": 125,
              "TableEnd": 129,
              "Alias": null,
              "Expr": {
                "Database": null,
                "Table": {
                  "Name": "cte1",
                  "QuoteType": 1,
                  "NamePos": 125,
                  "NameEnd": 129
                }
              },
              "HasFinal": false
            },
            "StatementEnd": 129,
            "SampleRatio": null,
            "HasFinal": false
          },
          "Right": {
            "Table": {
              "TablePos": 130,
              "TableEnd": 134,
              "Alias": null,
              "Expr": {
                "Database": null,
                "Table": {
                  "Name": "cte2",
                  "QuoteType": 1,
                  "NamePos": 130,
                  "NameEnd": 134
                }
              },
              "HasFinal": false
            },
            "StatementEnd": 134,
            "SampleRatio": null,
            "HasFinal": false
          },
          "Modifiers": null,
          "Constraints": null
        },
        "Modifiers": null,
        "Constraints": null
      }
    },
    "ArrayJoin": null,
    "Window": null,
    "Prewhere": null,
    "Where": null,
    "GroupBy": null,
    "WithTotal": false,
    "Having": null,
    "OrderBy": null,
    "LimitBy": null,
    "Limit": null,
    "Settings": null,
    "Format": null,
    "UnionAll": null,
    "UnionDistinct": null,
    "Except": null
  }
]